[IMP] l10n_be: update sales accounts in CoA as per ITAA guidelines

dijo-odoo opened the pull request:
### PURPOSE:
 - The first commit aims to **improve the Belgian Chart of Accounts** by cleaning up and restructuring sales-related income accounts to **align better with ITAA recommendations.**
 - The second commit aims to improve the accuracy of customer invoice filters by using **Account Type to income** instead of the internal group of type account, for accounts shown below each move line.

### SPECIFICATION:
- Removed/Replaced outdated income accounts from **700000 to 705200**.
- Set all income accounts from **708000 onwards** to be of type '**Other Income**'.
- Update the **fiscal position's mapping based** on new guidelines.
- Updated filters on customer invoices to use **account type = 'income'** instead of internal_group = 'income'.
- Please see reference [spreadsheet](https://tinyurl.com/y74n6e52)


task-4948805

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

hamo-odoo commented on addons/l10n_be/views/account_move_views.xml:
```diff
@@ -0,0 +1,21 @@
+        <field name="inherit_id" ref="account.view_move_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//page[@id='invoice_tab']//field[@name='invoice_line_ids']/list/field[@name='account_id']" position="attributes">
+                <attribute name="context">
+                    {   'partner_id': partner_id,
+                        'move_type': parent.move_type,
+                        'search_default_type_asset_fixed': 'in_' in parent.move_type,
+                        'search_default_type_expense': 'in_' in parent.move_type,
+                        'search_default_account_type': 'income',
+                        'preferred_account_type': 'out_' in parent.move_type and 'income' or 'in_' in parent.move_type and 'expense',
+                    }
+                </attribute>
```
Should it be generic or `l10n_be` specific? This might override the standard filter

RicGR98 commented on addons/l10n_be/views/account_move_views.xml:
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
```
This is added in in the first commit then deleted in the second ?

RicGR98 commented on addons/l10n_be/__manifest__.py:
```diff
@@ -46,6 +46,7 @@
         'data/account_tax_report_data.xml',
         'data/l10n_be_sequence_data.xml',
         'data/menuitem_data.xml',
+        'views/account_move_views.xml',
```
This is added in one commit and remove in the following one

RicGR98 reviewed (changes requested):
The PR is very confusing as the second commit modifies things from the first commit.

Couldn't they be merged ? The diff would be much cleaner as it's hard to follow what's being remove/added/modified

dijo-odoo commented:
> The PR is very confusing as the second commit modifies things from the first commit.
> 
> Couldn't they be merged ? The diff would be much cleaner as it's hard to follow what's being remove/added/modified
Hello @RicGR98 I mearged both commits into one , can you please take a look now !?
thanks

dylankiss reviewed (changes requested):
I just had a quick glance at the translations and noticed a lot of errors 🙈 
Can you please review them? If you have a list of your changes in a sheet e.g., I'm willing to have a look at it as well.

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a7005","Sales Co-Contractor","7005","income","","False","","Ventes co-contractant","Verkopen medecontractant","Verkäufe Vertragspartner","Sales and services performed in Belgium for which VAT is due by the co-contractor","Ventes et services effectuées en Belgique pour lesquelles la TVA est due par le cocontractant","Verkopen en diensten verricht in België waarvoor de btw verschuldigd is door de medecontractant","Verkäufe und Dienstleistungen in Belgien, für die der Vertragspartner die MwSt. schuldet","attn_VAT-OUT-00-CC",""
```

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a7006","Intra-Community Deliveries","7006","income","","False","","Ventes livraisons intracommunautaires","Intracommunautaire leveringen","Innergemeinschaftliche Lieferungen","Sales of goods with delivery within the European Union requiring reverse charge in the buyer's country","Ventes de biens avec livraison au sein de l'Union Européenne requérant l'autoliquidation dans le pays de l'acheteur","Verkoop van goederen met levering binnen de Europese Unie waarvoor verlegging van de btw in het land van de koper vereist is","Verkäufe von Waren mit Lieferung innerhalb der Europäischen Union, die ein Reverse-Charge-Verfahren im Land des Käufers erfordern","attn_VAT-IN-V82-00-EU-G",""
```

lman-odoo replied:
DE: Verkauf von Waren mit Lieferung innerhalb der Europäischen Union, die eine Umkehrung der Steuerschuldnerschaft im Land des Käufers erfordern

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a70061","Intra-Community Services","70061","income","","False","","Ventes services intracommunautaires","Intracommunautaire diensten","Innergemeinschaftliche Dienstleistungen","Services provided within the European Union requiring reverse charge in the buyer's country","Prestations de services au sein de l'Union Européenne requérant l'autoliquidation dans le pays de l'acheteur","Diensten verricht binnen de Europese Unie waarvoor verlegging van de btw in het land van de koper vereist is","Dienstleistungen innerhalb der Europäischen Union, die ein Reverse-Charge-Verfahren im Land des Käufers erfordern","attn_VAT-OUT-00-EU-S",""
```

lman-odoo replied:
DE: Innerhalb der Europäischen Union erbrachte Dienstleistungen, die eine Umkehrung der Steuerschuldnerschaft im Land des Käufers erfordern

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a7007","Non-EU Sales","7007","income","","False","","Ventes hors EU","Verkopen buiten de EU","Verkäufe außerhalb der EU","Sales to customers outside of the European Union","Ventes à des clients hors de l'Union européenne","Verkopen aan klanten buiten de Europese Unie","Verkäufe an Kunden außerhalb der Europäischen Union","attn_VAT-OUT-00-ROW",""
```

dylankiss reviewed (commented):
I took the liberty of proposing corrections for the changed account names (and adding missing German descriptions).

lman-odoo commented on addons/l10n_be/data/template/account.account-be.csv:
DE: Regulärer Verkauf von Waren in Belgien

lman-odoo commented on addons/l10n_be/data/template/account.account-be.csv:
To put it more concisely in DE: "Regulärer Verkauf von in Belgien erbrachten Dienstleistungen"

dylankiss commented:
@dijo-odoo Just discussed with @chklop as well. You can just copy-paste the changes from the suggestions I did here. And also add back the removed tags. Thank you 🙏

dijo-odoo commented:
> @dijo-odoo Just discussed with @chklop as well. You can just copy-paste the changes from the suggestions I did here. I updated the German also with the suggestions of Larissa and added back the removed tags. Thank you 🙏

Hello @dylankiss , can you please take a look at the updated changes..!?
thanks

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a7006","Intra-Community Deliveries","7006","income","account.account_tag_operating","False","","Ventes livraisons intracommunautaires","Intracommunautaire leveringen","Innergemeinschaftliche Lieferungen","Sales of goods with delivery within the European Union requiring reverse charge in the buyer's country","Ventes de biens avec livraison au sein de l'Union européenne requérant l'autoliquidation dans le pays de l'acheteur","Verkoop van goederen met levering binnen de Europese Unie waarvoor verlegging van de btw in het land van de koper vereist is","Verkauf von Waren mit Lieferung innerhalb der Europäischen Union, die eine Umkehrung der Steuerschuldnerschaft im Land des Käufers erfordern","attn_VAT-IN-V82-00-EU-G",""
```

dylankiss commented on addons/l10n_be/data/template/account.account-be.csv:
```suggestion
"a70061","Intra-Community Services","70061","income","account.account_tag_operating","False","","Ventes services intracommunautaires","Intracommunautaire diensten","Innergemeinschaftliche Dienstleistungen","Services provided within the European Union requiring reverse charge in the buyer's country","Prestations de services au sein de l'Union européenne requérant l'autoliquidation dans le pays de l'acheteur","Diensten verricht binnen de Europese Unie waarvoor verlegging van de btw in het land van de koper vereist is","Innerhalb der Europäischen Union erbrachte Dienstleistungen, die eine Umkehrung der Steuerschuldnerschaft im Land des Käufers erfordern","attn_VAT-OUT-00-EU-S",""
```

RicGR98 reviewed (approved):
Thanks all !

@robodoo r+

RicGR98 commented:
@robodoo r-

Could you rebase ?

dijo-odoo commented:
> @robodoo r-
> 
> Could you rebase ?

Done @RicGR98

h4818 commented on addons/account/views/account_move_views.xml:
```diff
@@ -1158,7 +1158,7 @@
                                                     'search_default_type_asset_fixed': 'in_' in parent.move_type,
                                                     'search_default_type_expense': 'in_' in parent.move_type,
-                                                    'search_default_incomeacc': 'out_' in parent.move_type,
+                                                    'search_default_account_type': 'income',
```
This seems wrong.
This view is used by both invoices and bills...
Why would bills have an 'income' account type as a search default?

dijo-odoo replied:
@h4818 ,
can you please take a look ? 
thanks

h4818 reviewed (approved):
Thanks @dijo-odoo 

@robodoo r+
